homeassistant/components/lyric/sensor.py

```diff
@@ -43,10 +42,84 @@
 
 
 @dataclass
-class LyricSensorEntityDescription(SensorEntityDescription):
+class LyricSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[LyricDevice], StateType | datetime]
+    suitable_fn: Callable[[LyricDevice], bool]
+
+
+@dataclass
+class LyricSensorEntityDescription(
+    SensorEntityDescription, LyricSensorEntityDescriptionMixin
+):
     """Class describing Honeywell Lyric sensor entities."""
 
-    value: Callable[[LyricDevice], StateType | datetime] = round
+
+DEVICE_SENSORS: list[LyricSensorEntityDescription] = [
+    LyricSensorEntityDescription(
+        key="indoor_temperature",
+        translation_key="indoor_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.indoorTemperature,
+        suitable_fn=lambda device: bool(device.indoorTemperature),
+    ),
+    LyricSensorEntityDescription(
+        key="indoor_humidity",
+        translation_key="indoor_humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda device: device.indoorHumidity,
+        suitable_fn=lambda device: bool(device.indoorHumidity),
+    ),
+    LyricSensorEntityDescription(
+        key="outdoor_temperature",
+        translation_key="outdoor_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.outdoorTemperature,
+        suitable_fn=lambda device: bool(device.outdoorTemperature),
+    ),
+    LyricSensorEntityDescription(
+        key="outdoor_humidity",
+        translation_key="outdoor_humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda device: device.displayedOutdoorHumidity,
+        suitable_fn=lambda device: bool(device.displayedOutdoorHumidity),
+    ),
+    LyricSensorEntityDescription(
+        key="next_period_time",
+        translation_key="next_period_time",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda device: get_datetime_from_future_time(
+            device.changeableValues.nextPeriodTime
+        ),
+        suitable_fn=lambda device: device.changeableValues
+        and device.changeableValues.nextPeriodTime,
+    ),
+    LyricSensorEntityDescription(
+        key="setpoint_status",
+        translation_key="setpoint_status",
+        icon="mdi:thermostat",
+        value_fn=lambda device: get_setpoint_status(
+            device.changeableValues.thermostatSetpointStatus,
+            device.changeableValues.nextPeriodTime,
+        ),
+        suitable_fn=lambda device: device.changeableValues
+        and device.changeableValues.thermostatSetpointStatus,
```

Comment:
Also, this logic comes from the old code, we only check if device.changeableValues.thermostatSetpointStatus is there, but we also use device.changeableValues.nextPeriodTime in our value function, is nextPeriodTime always there when thermostatSetpointStatus  is known?